src/vs/workbench/api/common/extHostTypeConverters.ts

```diff
@@ -857,14 +862,13 @@ export namespace CompletionItem {
 		if (suggestion.additionalTextEdits) {
 			const validAdditionalEdits = [];
 			for (const additionalEdit of suggestion.additionalTextEdits) {
-				const range = convertRange(additionalEdit.range);
-				if (range instanceof types.Range) {
-					validAdditionalEdits.push(new types.TextEdit(range, additionalEdit.text));
+				if (TextEdit.isModeTextEdit(additionalEdit) && convertRange(additionalEdit.range) instanceof types.Range) {
+					validAdditionalEdits.push(TextEdit.to(additionalEdit));
```

Comment:
Why the `if`-defence? Shouldn't the type always be correct?

Reply:
The edits.text of type ISingleEditOperation are possibly null, while modes.TextEdit requires a non-null string.

Is casting it to the type is sufficient?

Reply:
Yeah - cast it. It's TextEdit-objects that have sourced them, so they should be well defined.

Reply:
Renove the range check as well? Or keep it.

Reply:
nuke it